SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -51,29 +58,90 @@ QueryConstraint InterestFactory::CreateClientCheckoutRadiusConstraint(USpatialCl
 	//   - Other than the default from AActor, all radius constraints also include Component constraints to
 	//     capture specific types, including all derived types of that actor.
 
-	QueryConstraint CheckoutRadiusConstraint;
+	const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>();
+	QueryConstraint CheckoutRadiusConstraintRoot;
+	CheckoutConstraints.Empty();
 
-	CheckoutRadiusConstraint.OrConstraint.Add(CheckoutRadiusConstraintUtils::GetDefaultCheckoutRadiusConstraint());
+	if (!SpatialGDKSettings->bEnableNetCullDistanceInterest)
```

Comment:
This function is now very long- could each of the blocks under an if/else (i.e this one, then if net cull is enabled but frequency isn't, and then full frequency) be split out into their own functions that returns the list of constraints to add to the top level or(checkoutconstraintradiusroot)? Then this could become more of a switch statement based on the interest settings state, imo makes it easier to read

Reply:
Refactored - PTAL